src/signers/LightwalletSigner.ts

```diff
@@ -0,0 +1,87 @@
+
+import { TxSigner } from './TxSigner'
+import { User } from '../User'
+import { Utils } from '../Utils'
+import {
+  TxInfos,
+  TxInfosRaw,
+  RawTxObject
+} from '../typings'
+
+import { BigNumber } from 'bignumber.js'
+import * as lightwallet from 'eth-lightwallet'
+// declare let lightwallet
+
+/**
+ * The LightwalletSigner class contains functions for signing transactions with eth-lightwallet.
+ */
+export class LightwalletSigner implements TxSigner {
+  private _user: User
+  private _utils: Utils
+
+  constructor (user: User, utils: Utils) {
```

Comment:
This is the wrong dependency direction. User should depend on the signer, not the other way arround. If a user wants to sign, he needs a signing service, not the other way around. Then you should not have the problem why you needed to refactor the onboarding stuff.